receiver/pulsarreceiver/config.go

```diff
@@ -0,0 +1,148 @@
+	// "otlp_metrics" for metrics, "otlp_logs" for logs)
+	Topic string `mapstructure:"topic"`
+	// The Subscription that receiver will be consuming messages from (default "otlp_subscription")
+	Subscription string `mapstructure:"subscription"`
+	// Encoding of the messages (default "otlp_proto")
+	Encoding string `mapstructure:"encoding"`
+	// Name specifies the consumer name.
+	ConsumerName string `mapstructure:"consumer_name"`
+	// Set the path to the trusted TLS certificate file
+	TLSTrustCertsFilePath string `mapstructure:"tls_trust_certs_file_path"`
+	// Configure whether the Pulsar client accept untrusted TLS certificate from broker (default: false)
+	Insecure       bool           `mapstructure:"insecure"`
+	Authentication Authentication `mapstructure:"auth"`
+}
+
+type Authentication struct {
+	TLS    *TLS    `mapstructure:"tls"`
+	Token  *Token  `mapstructure:"Token"`
+	Athenz *Athenz `mapstructure:"athenz"`
+	OAuth2 *OAuth2 `mapstructure:"oauth2"`
+}
+
+type TLS struct {
+	CertFile string `mapstructure:"cert_file"`
+	KeyFile  string `mapstructure:"key_file"`
+}
+
+type Token struct {
+	Token string `mapstructure:"Token"`
+}
+
+type Athenz struct {
+	ProviderDomain  string `mapstructure:"provider_domain"`
+	TenantDomain    string `mapstructure:"tenant_domain"`
+	TenantService   string `mapstructure:"tenant_service"`
+	PrivateKey      string `mapstructure:"private_key"`
+	KeyID           string `mapstructure:"key_id"`
+	PrincipalHeader string `mapstructure:"principal_header"`
+	ZtsURL          string `mapstructure:"zts_url"`
+}
+
+type OAuth2 struct {
+	IssuerURL string `mapstructure:"issuer_url"`
+	ClientID  string `mapstructure:"client_id"`
+	Audience  string `mapstructure:"audience"`
+}
+
+var _ config.Receiver = (*Config)(nil)
+
+// Validate checks the receiver configuration is valid
+func (cfg *Config) Validate() error {
+	return nil
+}
+
+func (cfg *Config) auth() pulsar.Authentication {
+	authentication := cfg.Authentication
+	if authentication.TLS != nil {
+		return pulsar.NewAuthenticationTLS(authentication.TLS.CertFile, authentication.TLS.KeyFile)
+	}
+	if authentication.Token != nil {
+		return pulsar.NewAuthenticationToken(authentication.Token.Token)
+	}
+	if authentication.OAuth2 != nil {
+		return pulsar.NewAuthenticationOAuth2(map[string]string{
+			"issuerUrl": authentication.OAuth2.IssuerURL,
+			"clientId":  authentication.OAuth2.ClientID,
+			"audience":  authentication.OAuth2.Audience,
+		})
+	}
+	if authentication.Athenz != nil {
+		return pulsar.NewAuthenticationAthenz(map[string]string{
+			"providerDomain":  authentication.Athenz.ProviderDomain,
+			"tenantDomain":    authentication.Athenz.TenantDomain,
+			"tenantService":   authentication.Athenz.TenantService,
+			"privateKey":      authentication.Athenz.PrivateKey,
+			"keyId":           authentication.Athenz.KeyID,
+			"principalHeader": authentication.Athenz.PrincipalHeader,
+			"ztsUrl":          authentication.Athenz.ZtsURL,
+		})
+	}
+
+	return nil
+}
```

Comment:
This and other logic related to configuration and pulsar connection is pretty much the same between receiver and exporter. Can you please move the shared logic to `internal/pulsar` and reuse it across exporter and receiver?

Reply:
Can you please make the move from exporter to internal as a separate RP?

Reply:
I agree and I'll create another PR to move these logics to `internal/pulsar` after this PR is merged.